internal/static/static.go

```diff
@@ -0,0 +1,70 @@
+package static
+
+import (
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"golang.org/x/tools/go/packages"
+	"golang.org/x/tools/go/ssa"
+	"golang.org/x/tools/go/ssa/ssautil"
+)
+
+const (
+	// dogmaPkgPath is the full path of dogma package.
+	dogmaPkgPath = "github.com/dogmatiq/dogma"
+)
+
+// FromPackages loads the list of configkit.Application from the given list
+// of packages.Package items.
+func FromPackages(pkgs []*packages.Package) []configkit.Application {
+	prog, packages := ssautil.AllPackages(pkgs, ssa.SanityCheckFunctions)
+	prog.Build()
+
+	return fromSSAPackages(packages)
+}
+
+// fromSSAPackages loads the list of configkit.Application from the given list
+// of ssa.Package items.
+func fromSSAPackages(pkgs []*ssa.Package) []configkit.Application {
+	var result []configkit.Application
+
+	for _, pkg := range pkgs {
+		// The package can be returned as nil by ssautil.AllPackages() if it
+		// previously contained errors in types.Package.Errors.
+		if pkg == nil {
+			continue
+		}
+
+		if p := pkg.Prog.ImportedPackage(dogmaPkgPath); p != nil {
```

Comment:
I *think* you could do this once in `FromPackages()` if `pkg.Prog` is the same `*ssa.Program` as returned by `AllPackages()`.

Reply:
Yep, totally agree, I refactored that logic slightly in https://github.com/dogmatiq/configkit/pull/118/commits/d94b13badaa54f24a16e6caacd002cb88592384e, please LMK if this is something that you thought it would look like.